### PR TITLE
Close `Xml` result file as soon as possible

### PR DESCRIPTION
**Pull request by dougbu**
- #1515

### CODE REVIEW

**Comment by dougbu** [comment]
This PR is incomplete because I'm not sure how to test the change. GC can happen at any time, making the original problem difficult to reproduce in a unit test -- especially when other tests are running in the same process. Suggestions?

File: src/xunit.runner.msbuild/xunit.cs [hunk 1]
```diff
@@ -200,7 +200,8 @@ public override bool Execute()
             if (NeedsXml)
             {
                 if (Xml != null)
-                    assembliesElement.Save(new FileStream(Xml.GetMetadata("FullPath"), FileMode.OpenOrCreate));
+                    using (var xmlStream = new FileStream(Xml.GetMetadata("FullPath"), FileMode.OpenOrCreate, FileAccess.Write))
```
Comment (dougbu) [inline src/xunit.runner.msbuild/xunit.cs]: I'm addressing a nit: The `FileStream` constructor defaults to `FileAccess.ReadWrite` and that's not necessary here. But, passing the additional argument isn't mandatory.

**Comment by bradwilson** [comment]
I'm fine accepting this without a unit test.

**Comment by bradwilson** [comment]
Thanks!

**Pull request merged**